C/plugins/storage/postgres/connection.cpp

```
@@ -3312,3 +3314,712 @@ const char *p;
+	unsigned long rowsAffectedLastCommand = 0;
+	std::unordered_map<std::string, std::unordered_set<std::string> > columnMapFromDB;
+        std::unordered_map<std::string, std::vector<std::string> > indexMapFromDB;
+	std::unordered_map<std::string, std::string> tbKeyMapFromDB;
+	bool schemaCreationReq = false;
+
+	try 
+	{
+                if (payload.empty())
+                {
+			Logger::getLogger()->error("payload empty");
+                        return -1;
+                }
+                else
+                {
+                        if (document.Parse(payload.c_str()).HasParseError())
+                        {
+				Logger::getLogger()->error("Failed to parse JSON payload");
+                                return -1;
+                        }
+			if (!document.HasMember("schema"))
+                        {
+				Logger::getLogger()->error("schema absent from payload");
+				return -1;
+			}
+			else
+			{
+				schema = document["schema"].GetString();
+
+				if (!document.HasMember("service"))
+				{
+					Logger::getLogger()->error("service absent from payload");
+                                        return -1;
+				}
+				std::string service = document["service"].GetString();	
+				if (service.empty())
+				{
+					Logger::getLogger()->error("empty service name");
+                                        return -1;
+				}
+
+				if (!document.HasMember("version"))
+                        	{
+					Logger::getLogger()->error("version absent from payload");
+                                	return -1;
+                        	}
+				else
+				{
+					version = document["version"].GetInt();
+
+					if (!schema.empty())
+					{
+						std::string results;
+						if (findSchemaFromDB(service, schema, results))
+						{
+							if (!parseDatabaseStorageSchema(version, results, columnMapFromDB, indexMapFromDB, tbKeyMapFromDB, schemaCreationReq))
+							{
+								Logger::getLogger()->error("error in parsing Database Storage schema");
```

Again report names of the schema and service and what the parse error was - this might mean you need to report the error in the called function rather than here.

---

Fixed